File: app/src/main/java/com/jann_luellmann/thekenapp/data/view_model/EventViewModel.java
package com.jann_luellmann.thekenapp.data.view_model;

import com.jann_luellmann.thekenapp.data.model.Event;

import java.util.List;

import androidx.lifecycle.LiveData;

public class EventViewModel extends BaseViewModel<Event> {

    public EventViewModel() {
        super();
    }

    public void insert(Event event) {
        executor.execute(() -> db.eventDAO().insertAll(event));
    }

    public void insert(List<Event> events) {
        executor.execute(() -> db.eventDAO().insertAll(events.toArray(new Event[0])));
    }

    public void update(Event event) {
        executor.execute(() -> db.eventDAO().update(event));
    }

    @Override
    public LiveData<List<Event>> findAll() {
        return db.eventDAO().getAll();
    }

    @Override
    public void insert(long eventId, Event event) {
        insert(event);
    }

    @Override
    public void delete(Event event) {
        executor.execute(() -> db.eventDAO().delete(event));
    }
}

File: app/src/main/java/com/jann_luellmann/thekenapp/data/view_model/EventViewModel.kt
package com.jann_luellmann.thekenapp.data.view_model

import androidx.lifecycle.LiveData
import com.jann_luellmann.thekenapp.data.model.Event

class EventViewModel : BaseViewModel<Event>() {
    fun insert(event: Event): Long {
        return db.eventDAO().insert(event)
    }

    fun insert(events: List<Event>) {
        executor.execute { db.eventDAO().insertAll(*events.toTypedArray()) }
    }

    fun update(event: Event) {
        executor.execute { db.eventDAO().update(event) }
    }

    override fun findAll(): LiveData<List<Event>> {
        return db.eventDAO().all
    }

    override fun insert(eventId: Long, event: Event) {
        insert(event)
    }

    override fun delete(event: Event) {
        executor.execute {
            db.eventDAO().delete(event)
            db.eventWithDrinksAndCustomersDAO().deleteEventCustomers(event.eventId)
            db.eventWithDrinksAndCustomersDAO().deleteEventDrinks(event.eventId)
            db.eventAndCustomerWithDrinksDAO().delete(event.eventId)
        }
    }
}